### PR TITLE
Extend precision audit to fixtures 67-82 (`RaggedTensor.from_*` family)

### PR DESCRIPTION
**Pull request by khatchad**
## Summary

- Extend the two-layer precision audit to fixtures 67-82, covering the full `tf.RaggedTensor.from_*` constructor family (`from_nested_value_rowids`, `from_row_lengths`, `from_row_limits`, `from_row_splits`, `from_row_starts`, `from_value_rowids`).
- Static-import `STRING` and shift to `Arrays.asList` per the project's no-FQN style; sweep applies to previously-inlined fixture-59 / 60-66 anchors as well.

## Findings

**Layer 1 (Ariadne, consumed 0.44.0):**

- `from_nested_value_rowids` over a `tf.keras.Input(shape=[None], dtype=tf.string)` (67-70): `STRING, (NumericDim(4), null, null, null)` — `NumericDim` from `max(rowids[0]) + 1`, two ragged-marker `null` entries, one dynamic-dim `null` from the `keras.Input` shape.
- Single-ragged `from_*` constructors (71-82): `INT32, (NumericDim(nrows), null)` — `nrows = 5` for 71-79 (`from_row_lengths`/`limits`/`splits`/`starts`), `nrows = 4` for 80-82 (`from_value_rowids` over `[0,0,0,0,2,2,2,3]`).

**Layer 2 (Hybridize):** every `null` collapses to `SymbolicDim("?")`. Correct semantics for the dynamic dim (a `TensorSpec` wildcard genuinely matches runtime behavior) but lossy for ragged dims (a `RaggedTensorSpec` is the precise emission—#524).

The TODO anchors at `testHasLikelyTensorParameter59` and at the translation site in `Function#inferInputSignature` continue to carry the flip references for wala/ML#544 (typed `RaggedDim`, fixed on Ariadne master via ponder-lab/ML#320 but not yet in a Hybridize-consumed release) and #524.

## Test Plan

- [x] `./mvnw verify -pl edu.cuny.hunter.hybridize.tests` — 492 tests, 0 failures, 11 skipped
- [ ] CI green
- [ ] Copilot threads clean

## Cross-Refs

- wala/ML#544—typed `RaggedDim` sentinel (Layer-1 flip target; fixed on Ariadne master).
- wala/ML#545—typed `DynamicDim` sentinel (Layer-1 flip target for the dynamic-dim positions in 67-70).
- #524—`RaggedTensorSpec` emission (Layer-2 flip target).
- #525—batch-2 (`from_nested_row_splits` fixtures 60-66).
- #522—batch-1 (fixtures 1-58 + 59).

### CODE REVIEW

File: edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java [hunk 1]
```diff
@@ -3834,7 +3835,7 @@ public void testHasLikelyTensorParameter59() throws Exception {
 		// Precision audit. `tf.RaggedTensor.from_nested_row_splits(values, splits)` — runtime dtype=int32, shape=(3, None, None).
 		// Ariadne emits an INT32 TensorType with three dims: a known constant 3 followed by two raggedness markers (raw null).
 		// TODO(wala/ML#544): flip the `null` entries to `new RaggedDim()` once Ariadne ships the typed sentinel.
-		TensorType ariadne = new TensorType(INT32, java.util.Arrays.asList(new NumericDim(3), null, null));
+		TensorType ariadne = new TensorType(INT32, Arrays.asList(new NumericDim(3), null, null));
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java:3845]: `import java.util.Arrays;` already exists at line 48; the simple-name form compiles. `./mvnw verify` passes locally on this branch.

File: edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java [hunk 2]
```diff
@@ -3915,127 +3916,143 @@ public void testHasLikelyTensorParameter66() throws Exception {
 	 */
 	@Test
 	public void testHasLikelyTensorParameter67() throws Exception {
-		testHasLikelyTensorParameterHelper();
+		testHasLikelyTensorParameterHelper(new TensorType(STRING, Arrays.asList(new NumericDim(4), null, null, null)),
+				new TensorType(STRING, List.of(new NumericDim(4), new SymbolicDim("?"), new SymbolicDim("?"), new SymbolicDim("?"))));
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java:3927]: Same DRY suggestion was raised and declined on #525 with the audit-driver rationale: the constants encode a current analysis gap (wala/ML#544 / #545 / #524) that will not survive the upstream flips, naming by value is awkward enough to signal the abstraction is wrong, and inline duplication is documentary—a reader sees the same imprecise pattern repeated across the family, which IS the finding. Carrying the same answer forward here.

File: edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java [hunk 3]
```diff
@@ -3915,127 +3916,143 @@ public void testHasLikelyTensorParameter66() throws Exception {
...
 	}
 
 	/**
 	 * Test for #2 for TF API `RaggedTensor.from_nested_value_rowids`.
 	 */
 	@Test
 	public void testHasLikelyTensorParameter68() throws Exception {
-		testHasLikelyTensorParameterHelper();
+		testHasLikelyTensorParameterHelper(new TensorType(STRING, Arrays.asList(new NumericDim(4), null, null, null)),
+				new TensorType(STRING, List.of(new NumericDim(4), new SymbolicDim("?"), new SymbolicDim("?"), new SymbolicDim("?"))));
```
[the remark this answers is not in the document]
Comment (khatchad) [reply edu.cuny.hunter.hybridize.tests/test cases/edu/cuny/hunter/hybridize/tests/HybridizeFunctionRefactoringTest.java:3936]: Same DRY suggestion was raised and declined on #525 with the audit-driver rationale: the constants encode a current analysis gap (wala/ML#544 / #545 / #524) that will not survive the upstream flips, naming by value is awkward enough to signal the abstraction is wrong, and inline duplication is documentary—a reader sees the same imprecise pattern repeated across the family, which IS the finding. Carrying the same answer forward here.
